### ISSUE TITLE
remove New() -> func pattern

### ISSUE THREAD

**Issue by tj** [gohttp/app#10]
thought it made sense before but meh, nicer to just make things http.Handler's and chain them manually

**Comment by tarrsalah** [comment]
Yeah, Unnecessary glue code, chaining `middleware(http.Handler) http.Handler` middlewares manually like `m1(m2(m3(handler)`  seems to be fine for now.
